packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -707,16 +565,57 @@ contract Bridge is EssentialContract, IBridge {
         }
     }
 
-    function _isPostInvocationDelay(
-        uint256 _receivedAt,
-        uint256 _invocationDelay
+    function _calcFee(
+        uint256 _msgFee,
+        uint256 _msgGasLimit,
+        uint256 _gasUsed
     )
         private
         view
-        returns (bool)
+        returns (uint256)
+    {
+        if (_msgFee == 0 || _msgGasLimit == 0) return 0;
+
+        uint256 maxFee = _msgFee * _gasUsed / _msgGasLimit;
+        uint256 baseFee = block.basefee * _gasUsed;
+        return _msgFee.min(baseFee >= maxFee ? maxFee : (maxFee + baseFee) >> 1);
```

Comment:
Just throwing it in here (helps with understanding)
Example 1.:

Let's say `maxFee` is 99.
Let's say `baseFee` is 100.

In this case the returned value (fee) is 99 - while the baseFee (so the actual, minimum real cost) would be higher (`100`), no ? This fee is then sent to `msg.sender` (aka `relayer`) -> But since the actual cost is higher, `relayer` will not process.

Example 2.:
Let's say `maxFee` is 110.
Let's say `baseFee` is 100.

In this case, the fee would be `110` `MIN` `((110+100) / 2 = 105 )` --> 105, so the relayer would process.

This part looks about good to me.

Reply:
In example one, I expect the relayer NOT to process the message.